_posts/2022-01-10-reassign-partitions.md

```diff
@@ -0,0 +1,420 @@
+
+While using the tool, you have to provide it with two essential JSON files: `topics.json` and `reassignment.json`.
+Wondering what these two JSON files really do? Let's talk a bit about them:
+
+- The `topic.json` basically consists of the topics that we want to reassign or move. Based on this JSON file, our tool will generate a proposal `reassignment.json` that we can use directly or modify further.
+
+- The `reassignment.json` file is a configuration file that is used during the partition reassignment process. The reassignment partition tool will generate a proposal `reassignment.json` file based on a `topics.json` file. You can change the `reassignment.json` file as per your requirement and use it.
+
+It has three different phases:
+
+| Phases | Description |
+| :-----------: | ------------- |
+| `generate`  | Takes a set of topics and brokers and generates a reassignment JSON file which will result in the partitions of those topics being assigned to those brokers. Because this operates on whole topics, it cannot be used when you only want to reassign some partitions for some topics.|
+| `execute` | Takes a reassignment JSON file and applies it to the partitions and brokers in the cluster. The `reassignment.json` file can be either the one proposed by the `--generate` command or written by the user itself. |
+| `verify`  | Using the same reassignment JSON file as the `--execute` step, `--verify` checks whether all the partitions in the file have been moved to their intended brokers. If the reassignment is complete, `--verify` also removes any traffic throttles (`--throttle`) that are in effect. Unless removed, throttles will continue to affect the cluster even after the reassignment has finished. |
+
+## Why use the Kafka reassignment partition tool?
+
+The Kafka reassignment partition tool can help you address a variety of use cases.
+
+Some of these are listed here:
+
+1. You can reassign the partition between the brokers anytime. For example, it can be used at times when you want to scale down the number of brokers in your cluster.
+   You can assign partitions from the broker to be scaled down to other brokers which will handle these partitions now.
+
+2. With the help of this tool, you can increase the no. of partitions / replicas which can help with increasing the throughput of the topic.
+
+## Example time
+
+Let us understand the working of this tool with an interesting problem
+Suppose we have 5 Kafka Brokers and after looking at the partition details we get to realize that the brokers are not too busy, so we can scale them down to 3.
+Through this example we will take a look at how the three phases of the Kafka reassignment partition tool(`--generate`, `--execute` and `--verify`) works.
+We will generate the JSON data that will be used in the `reassignment.json` file.
+We will then scale down the Kafka cluster using the generated `reassignment.json` file.
```

Comment:
This sentence is misleading I think. You are not scaling down the Kafka cluster using a reassignment JSON file, what did you mean really?